CONTRIBUTING.md

```diff
@@ -0,0 +1,258 @@
+Not even our project lead is exempt from reviews and RFCs!
+By giving feedback on this work (and related supporting work), you can help us make sure our releases are both high-quality and timely.
+
+Finally, if nothing brings you more satisfaction than seeing every last issue tagged and all resolved issues closed, feel free to message @cart for a Bevy org role to help us keep things tidy.
+As discussed in [*How we're organized*](https://github.com/bevy-engine/bevy/blob/main/CONTRIBUTING.md#how-were-organized), this is not intended to have a high bar.
+
+### Contributing your own ideas
+
+Bevy is actively open to new ideas and serious contributions from outside community members.
+If you're new to Bevy, here's the workflow we use:
+
+1. Fork the `bevyengine/bevy` repository on GitHub, you'll need to create a GitHub account if you don't have one already.
+2. Make your changes in a local clone of your fork, typically in its own new branch.
+   1. Try to split your work into separate commits, each with a new purpose. Be particularly thoughtful of this when responding to reviews so we can see what's changed.
+3. For a higher chance of CI passing the first time, consider locally running `cargo run -p ci`. You can run the commands manually:
+    1. `cargo fmt --all -- --check` (remove `--check` to let the command fix found problems)
+    2. `cargo clippy --workspace --all-targets --all-features -- -D warnings -A clippy::type_complexity`
+    3. `cargo test --all-targets --workspace`
+4. When working with Markdown (`.md`) files, Bevy's CI will check markdown files like this one using [markdownlint](https://github.com/DavidAnson/markdownlint).
+To locally lint your files using the same workflow as our CI:
+   1. Install [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli).
+   2. Run `markdownlint -f -c .github/linters/.markdown-lint.yml .` in the root directory of the Bevy project.
+5. Push your changes to your fork and open a Pull Request.
+6. If you're a first time contributor to this repo, @cart [will need to manually trigger CI for your PR](https://github.blog/changelog/2021-04-22-github-actions-maintainers-must-approve-first-time-contributor-workflow-runs/). Feel free to ping him for this.
+7. Respond to any CI failures or review feedback. While CI failures must be fixed before we can merge your PR; you do not need to *agree* with all feedback from your reviews, merely acknowledge that it was given. If you cannot come to an agreement, leave the thread open and defer to @cart's final judgement.
+8. When your PR is ready to merge, @cart will review it and suggest final changes. If those changes are minimal he may even apply them directly to speed up merging.
+
+If you end up creating a new crate to the `bevy` repo:
+
+1. Add a "Bevy Contributors <bevyengine@gmail.com>" entry in the Author field of `Cargo.toml`.
+2. Add the new crate to the ./tools./publish.sh file.
```

Comment:
I'm not sure about the stance of certain licenses, but IMO avoiding LGPL (v3) licenses would be best.
Since they require anyone who uses them to open-source their source code, and then would possible prevent some groups from using bevy.

And if so, we should leave a comment here about specific licenses?

Reply:
This will be for subcrates of Bevy itself, which are implicitly part of the same license.

For ecosystem crates, the plugin guidelines discusses our preference for non-permissive licenses.

Reply:
I'm confused: I thought our preference was for permissive licences?